### PR TITLE
Completely replace Travis and AppVeyor with GitHub Actions

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Completely replace Travis and AppVeyor with GitHub Actions.

Removes any references to Travis and AppVeyor and replaces them with a generic "CI".

## Why It's Good For The Game
1. **SPEED**

Here's a random Travis build I pulled:

image

GitHub Actions completes all these tasks in about 5 minutes tops. It also doesn't have to wait for other commits to finish building, unlike Travis.

2. Reliability

Travis is down, literally *right now*. For the past day. GitHub Actions is consistently reliable.

3. Flexible

GitHub Actions is easier to interface with than Travis and AppVeyor. I'd like to create a merge bot, and it's much smoother to get it to work with GitHub Actions than it is with Travis.

## Concerning GitHub

I acknowledge that we have a bit of a tense relationship with GitHub, and we're taking some precautions in case the repository goes down again and we have to move.

In the event that this happens, I've been told GitLab has a similar CI configuration to Actions, but if all else fails--https://github.com/nektos/act is available to run these locally.

### CODE REVIEW

**Comment by optimumtact** [comment]
```
In the event that this happens, I've been told GitLab has a similar CI configuration to Actions, but if all else fails--https://github.com/nektos/act is available to run these locally.
```
He lied confidently

**Comment by ZeWaka** [comment]
based

File: .github/workflows/generate_documentation.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        run: |
+          ~/dmdoc
+          touch dmdoc/.nojekyll
+          echo codedocs.tgstation13.org > dmdoc/CNAME
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@3.7.1
+        with:
+          BRANCH: gh-pages
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          FOLDER: dmdoc
```
Comment (ZeWaka) [inline .github/workflows/generate_documentation.yml:28]: consider CLEAN

Comment (Jared-Fogle) [reply .github/workflows/generate_documentation.yml:28]: I saw this as an option but didn't quite understand the point, can you fill me in?

Comment (ZeWaka) [reply .github/workflows/generate_documentation.yml:28]: You'd also use it with SINGLE_COMMIT likely. Basically, it ensures that the repository is only stuff generated by the script, and ensures files don't stick around that aren't needed anymore. I anticipated cache errors and stuff like that for Goon's, so I set it to both.

**Comment by Cyberboss** [comment]
Good PR

File: .github/workflows/generate_documentation.yml (hunk 1 above)
Comment (Jared-Fogle) [reply .github/workflows/generate_documentation.yml:28]: Ah, I see. Good suggestion!

**Comment by spookydonut** [comment]
gitlab CI is hobbled by paranoia.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        run: |
+          sudo dpkg --add-architecture i386
+          sudo apt update
+          sudo apt install libssl1.1:i386
+          bash tools/ci/install_rust_g.sh
+      - name: Compile and run tests
+        run: |
+          bash tools/ci/install_byond.sh
+          source $HOME/BYOND/byond/bin/byondsetup
+          bash tools/ci/dm.sh -DCIBUILDING tgstation.dme
```
Comment (TiviPlus) [inline .github/workflows/test.yml:84]: this

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        uses: actions/cache@v2
+        with:
+          path: $HOME/BYOND
+          key: ${{ runner.os }}-byond
+      - name: Compile All Maps
+        run: |
+          bash tools/ci/install_byond.sh
+          source $HOME/BYOND/byond/bin/byondsetup
+          python3 tools/ci/template_dm_generator.py
+          bash tools/ci/dm.sh -DCIBUILDING -DCITESTING -DALL_MAPS tgstation.dme
```
Comment (TiviPlus) [inline .github/workflows/test.yml:49]: that

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        with:
+          path: $HOME/BYOND
+          key: ${{ runner.os }}-byond
+      - name: Setup database
+        run: |
+          sudo systemctl start mysql
+          mysql -u root -proot -e 'CREATE DATABASE tg_ci;'
+          mysql -u root -proot tg_ci < SQL/tgstation_schema.sql
+          mysql -u root -proot -e 'CREATE DATABASE tg_ci_prefixed;'
+          mysql -u root -proot tg_ci_prefixed < SQL/tgstation_schema_prefixed.sql
```
Comment (TiviPlus) [inline .github/workflows/test.yml:73]: also this maybe but maybe not

**Comment by TiviPlus** [comment]
fuck wrong codebase

**Comment by Jared-Fogle** [comment]
all the downstreams want a piece

**Pull request merged**